pkg/operation/botanist/component/vpnshoot/vpnshoot.go

```diff
@@ -0,0 +1,515 @@
+type Values struct {
+	// Image is the container image used for vpnShoot.
+	Image string
+	// PodAnnotations is the set of additional annotations to be used for the pods.
+	PodAnnotations map[string]string
+	// vpaEnabled marks whether VerticalPodAutoscaler is enabled for the shoot
+	VPAEnabled bool
+	// reversedVPNEnabled marks whether ReversedVPN is enabled for the shoot
+	ReversedVPNEnabled bool
+	// PodNetworkCIDR is the CIDR of the pod network.
+	PodNetworkCIDR string
+	// ServiceNetworkCIDR is the CIDR of the node network.
+	ServiceNetworkCIDR string
+	// NodeNetworkCIDR is the CIDR of the node network.
+	NodeNetworkCIDR string
+	// ReversedVPNHeader is the header for the reversedVPN.
+	ReversedVPNHeader string
+	// ReversedVPNEndPoint is the end point for the reversedVPN.
+	ReversedVPNEndPoint string
+	// ReversedVPNOpenVPNPort is the port for the reversedVPN.
+	ReversedVPNOpenVPNPort string
+}
+
+// New creates a new instance of DeployWaiter for vpnshoot
+func New(
+	client client.Client,
+	namespace string,
+	values Values,
+) Interface {
+	return &vpnShoot{
+		client:    client,
+		namespace: namespace,
+		values:    values,
+	}
+}
+
+type vpnShoot struct {
+	client    client.Client
+	namespace string
+	values    Values
+	secrets   Secrets
+}
+
+func (v *vpnShoot) Deploy(ctx context.Context) error {
+	data, err := v.computeResourcesData()
+	if err != nil {
+		return err
+	}
+	return managedresources.CreateForShoot(ctx, v.client, v.namespace, ManagedResourceName, false, data)
+}
+
+func (v *vpnShoot) Destroy(ctx context.Context) error {
+	return managedresources.DeleteForShoot(ctx, v.client, v.namespace, ManagedResourceName)
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (v *vpnShoot) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, v.client, v.namespace, ManagedResourceName)
+}
+
+func (v *vpnShoot) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, v.client, v.namespace, ManagedResourceName)
+}
+
+func (v *vpnShoot) computeResourcesData() (map[string][]byte, error) {
+
+	var (
+		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		secretTLSAuth = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vpn-shoot-tlsauth",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: v.secrets.TLSAuth.Data,
+		}
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				//TODO(shafeeqes): what about for ReversedVPN not enabled case?
```

Comment:
What is the exact question here?

Reply:
was thinking whether the name will differ in the case of ReversedVPN enabled. like `vpn-shoot-client`